flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/PendingCheckpoint.java

```diff
@@ -107,7 +107,11 @@ public enum TaskAcknowledgeResult {
     /** The checkpoint properties. */
     private final CheckpointProperties props;
 
-    /** The promise to fulfill once the checkpoint has been completed. */
+    /**
+     * Because onCompletionPromise is not required to synchronize with the completion status of
+     * pendingCheckpoint. The promise to fulfill once the checkpoint has been completed and is added
+     * to the store.
+     */
```

Comment:
-> The promise to fulfill once the checkpoint has been completed. Note that it will be completed only after the checkpoint is successfully added to CompletedCheckpointStore.